### ISSUE TITLE
installperl patch related to $Config{dlext}

### ISSUE THREAD

**Issue by p5pRT** [Perl/perl5#14337]
Migrated from [rt.perl.org#123436](https://rt-archive.perl.org/perl5/Ticket/Display.html?id=123436) (status was 'resolved')

Searchable as RT123436$

**Comment by p5pRT** [comment]
### From @kmx

#### Created by @kmx



Please consider the following simple patch for installperl

&nbsp;      if \($Is\_Cygwin\) \{
&nbsp;      $perldll = $libperl;
&nbsp;      \} else \{
\-    $perldll = 'perl5'\.$Config\{patchlevel\}\.'\.'\.$dlext;
\+    $perldll = 'perl5'\.$Config\{patchlevel\}\.'\.'\.$so;
&nbsp;      \}

The patch is necessary if you want to build a perl with $Config\{dlext\}
different from $Config\{so\}\. Which is the case of strawberry perl
where $Config\{dlext\} is 'xs\.dll' whereas $Config\{so\} is 'dll'

Thanks

\-\-
kmx

<details><summary>Perl Info</summary>

```

Flags:
     category=core
     severity=low

Site configuration information for perl 5.20.1:

Configured by strawberry-perl at Mon Sep 15 13:28:28 2014.

Summary of my perl5 (revision 5 version 20 subversion 1) configuration:

   Platform:
     osname=MSWin32, osvers=6.3, archname=MSWin32-x64-multi-thread
     uname='Win32 strawberry-perl 5.20.1.1 #1 Mon Sep 15 13:26:45 2014 x64'
     config_args='undef'
     hint=recommended, useposix=true, d_sigaction=undef
     useithreads=define, usemultiplicity=define
     use64bitint=define, use64bitall=undef, uselongdouble=undef
     usemymalloc=n, bincompat5005=undef
   Compiler:
     cc='gcc', ccflags =' -s -O2 -DWIN32 -DWIN64 -DCONSERVATIVE 
-DPERL_TEXTMODE_SCRIPTS -DPERL_IMPLICIT_CONTEXT -DPERL_IMPLICIT_SYS 
-DUSE_PERLIO -fwrapv -fno-strict-aliasing -mms-bitfields',
     optimize='-s -O2',
     cppflags='-DWIN32'
     ccversion='', gccversion='4.8.3', gccosandvers=''
     intsize=4, longsize=4, ptrsize=8, doublesize=8, byteorder=12345678
     d_longlong=define, longlongsize=8, d_longdbl=define, longdblsize=12
     ivtype='long long', ivsize=8, nvtype='double', nvsize=8, Off_t='long 
long', lseeksize=8
     alignbytes=8, prototype=define
   Linker and Libraries:
     ld='g++.exe', ldflags ='-s -L"C:\tmp64\perl\lib\CORE" -L"C:\tmp64\c\lib"'
     libpth=C:\tmp64\c\lib C:\tmp64\c\x86_64-w64-mingw32\lib 
C:\tmp64\c\lib\gcc\x86_64-w64-mingw32\4.8.3
     libs=-lmoldname -lkernel32 -luser32 -lgdi32 -lwinspool -lcomdlg32 
-ladvapi32 -lshell32 -lole32 -loleaut32 -lnetapi32 -luuid -lws2_32 -lmpr 
-lwinmm -lversion -lodbc32 -lodbccp32 -lcomctl32
     perllibs=-lmoldname -lkernel32 -luser32 -lgdi32 -lwinspool -lcomdlg32 
-ladvapi32 -lshell32 -lole32 -loleaut32 -lnetapi32 -luuid -lws2_32 -lmpr 
-lwinmm -lversion -lodbc32 -lodbccp32 -lcomctl32
     libc=, so=dll, useshrplib=true, libperl=libperl520.a
     gnulibc_version=''
   Dynamic Linking:
     dlsrc=dl_win32.xs, dlext=xs.dll, d_dlsymun=undef, ccdlflags=' '
     cccdlflags=' ', lddlflags='-mdll -s -L"C:\tmp64\perl\lib\CORE" 
-L"C:\tmp64\c\lib"'



@INC for perl 5.20.1:
     C:/tmp64/perl/site/lib
     C:/tmp64/perl/vendor/lib
     C:/tmp64/perl/lib
     .


Environment for perl 5.20.1:
     HOME=C:\tmp64\data
     LANG (unset)
     LANGUAGE (unset)
     LD_LIBRARY_PATH (unset)
     LOGDIR (unset)
PATH=C:\tmp64\perl\site\bin;C:\tmp64\perl\bin;C:\tmp64\c\bin;C:\Windows\system32;C:\Windows;C:\Windows\System32\Wbem
     PERL_BADLANG (unset)
     SHELL (unset)


```
</details>

**Comment by p5pRT** [comment]
### From @steve-m-hay

On Tue Dec 16 00&#8203;:30&#8203;:40 2014\, kmxx wrote&#8203;:
> Please consider the following simple patch for installperl
> 
> if \($Is\_Cygwin\) \{
> $perldll = $libperl;
> \} else \{
> \-    $perldll = 'perl5'\.$Config\{patchlevel\}\.'\.'\.$dlext;
> \+    $perldll = 'perl5'\.$Config\{patchlevel\}\.'\.'\.$so;
> \}
> 
> The patch is necessary if you want to build a perl with $Config\{dlext\}
> different from $Config\{so\}\. Which is the case of strawberry perl
> where $Config\{dlext\} is 'xs\.dll' whereas $Config\{so\} is 'dll'
> 

Thanks\, applied in commit dc5c2addf88baf2341663222a7ef79d1b8df8429\.

\(This make good sense\, according to the definitions in Porting/Glossary\, regardless of whether the slightly more controversial \#117001 goes in or not\.\)

**Comment by p5pRT** [comment]
The RT System itself - Status changed from 'new' to 'open'

**Comment by p5pRT** [comment]
@steve-m-hay - Status changed from 'open' to 'pending release'

**Comment by p5pRT** [comment]
### From @khwilliamson

Thanks for submitting this ticket

The issue should be resolved with the release today of Perl v5\.22\, available at http&#8203;://www\.perl\.org/get\.html
If you find that the problem persists\, feel free to reopen this ticket

\-\- 
Karl Williamson for the Perl 5 porters team

**Comment by p5pRT** [comment]
@khwilliamson - Status changed from 'pending release' to 'resolved'
